vicon_mujoco_init/robot_init.py

```diff
@@ -0,0 +1,155 @@
+import numpy as np
+from scipy.spatial.transform import Rotation as R
+
+# to run this demo, start pam_mujoco with mujoco_id "vicon_robot_init"
+# pam_mujoco vicon_robot_init && python3 <pathToCode>/pam_demos/vicon_mujoco_init/robot_init.py
+
+# read a test vicon frame from file
+print('read a test vicon frame from file')
+vj = vicon_transformer.ViconJson()
+print(vj.j['subjectNames'])
+vj.zmq_disconnect()
+
+### translations
+# robot
+r_pos_shoulder = vj.get_robot_shoulder_pos()
+
+# table
+t_pos = vj.get_table_pos()
+
+### rotations
+# robot
+r_rot_xy = vj.get_robot_rot()
+
+# table
+t_rot_xy = vj.get_table_rot()
+
+# creating the mujoco's configuration, and getting the handle.
+# contrary to tutorial 1 to 3, the robot will be joint controlled (i.e.
+# position and velocity).
+robot_default = pam_mujoco.MujocoRobot("robot_default", control=pam_mujoco.MujocoRobot.JOINT_CONTROL)
+robot_vicon_init = pam_mujoco.MujocoRobot("robot_vicon_init", 
+                                        control=pam_mujoco.MujocoRobot.JOINT_CONTROL,
+                                        position=r_pos_shoulder,
+                                        orientation=r_rot_xy)
+
+table = pam_mujoco.MujocoTable("table_vicon_init",
+                                position = t_pos,
+                                orientation = t_rot_xy)
+ball = pam_mujoco.MujocoItem(
+    "ball", control=pam_mujoco.MujocoItem.CONSTANT_CONTROL, color=(1, 0, 0, 1)
+)
+
+hit_point = pam_mujoco.MujocoItem(
+    "hit_point", control=pam_mujoco.MujocoItem.CONSTANT_CONTROL
+)
+graphics = True
+accelerated_time = False
+handle = pam_mujoco.MujocoHandle(
+    "vicon_robot_init",
+    table=table,
+    robot1=robot_default, 
+    robot2=robot_vicon_init,
+    balls=(ball,),
+    hit_points=(hit_point,),
+    graphics=graphics,
+    accelerated_time=accelerated_time,
+)
+
+# getting the frontend connected to the robot's pressure controller
+robot_default = handle.frontends["robot_default"]
+robot_vicon_init = handle.frontends["robot_vicon_init"]
+ball = handle.frontends["ball"]
+hit_point = handle.frontends["hit_point"]
+
+# moving the balls and the robot to a target position
+
+print("starting state ball: {}".format(ball.latest()))
+
+
+# target position of balls
+position1 = (0.5, 3, 1)
+position2 = (1.5, 3, 1)
+velocity = (0, 0, 0)
+# target position of robot
+joints = (math.pi / 4.0, -math.pi / 4.0, math.pi / 4.0, -math.pi / 4.0)
+joint_velocities = (0, 0, 0, 0)
+
+duration = o80.Duration_us.seconds(2)
+
+ball.add_command(position1, velocity, duration, o80.Mode.QUEUE)
+robot_default.add_command(joints, joint_velocities, duration, o80.Mode.QUEUE)
+robot_vicon_init.add_command(joints, joint_velocities, duration, o80.Mode.QUEUE)
+
+ball.pulse()
+robot_default.pulse()
+robot_vicon_init.pulse()
+
+
+print("reached state ball: {}".format(ball.latest()))
+print("reached state robot: {}".format(robot_default.latest()))
+print("reached state robot: {}".format(robot_vicon_init.latest()))
```

Comment:
Looks like the prints above display the state of the ball / robot / robot vicon at the beginning of the motion, not at the end ?